io-engine-tests/src/file_io.rs

```diff
@@ -13,22 +13,41 @@ pub fn set_test_buf_rng_seed(seed: u64) {
     RNG_SEED.set(ChaCha8Rng::seed_from_u64(seed)).unwrap();
 }
 
-fn create_test_buf(size_mb: usize) -> Vec<u8> {
+fn create_test_buf(buf_size: BufferSize) -> Vec<u8> {
     let rng = RNG_SEED.get_or_init(ChaCha8Rng::from_entropy).clone();
 
     let range = Uniform::new_inclusive(0, u8::MAX);
 
     rng.sample_iter(&range)
-        .take(size_mb * 1024 * 1024)
+        .take(buf_size.size_bytes())
         .collect()
 }
 
+pub enum BufferSize {
+    Bytes(usize),
+    Blocks(usize),
+    Mb(usize),
+    Kb(usize),
+}
+
+impl BufferSize {
+    pub fn size_bytes(&self) -> usize {
+        match &self {
+            BufferSize::Bytes(s) => *s,
+            BufferSize::Blocks(s) => s * 512,
```

Comment:
Just curious: can't we have devices with 4K blocks ?

Reply:
I think so yes, if not then I'd consider it a bug :)